numpy-stubs/__init__.pyi

```diff
@@ -792,3 +797,124 @@ class AxisError(ValueError, IndexError):
     def __init__(
         self, axis: int, ndim: Optional[int] = ..., msg_prefix: Optional[str] = ...
     ) -> None: ...
+
+# Functions from np.core.fromnumeric
+_Mode = Literal["raise", "wrap", "clip"]
+_Order = Literal["C", "F", "A"]
+_PartitionKind = Literal["introselect"]
+_SortKind = Literal["quicksort", "mergesort", "heapsort", "stable"]
+
+_Generic = TypeVar("_Generic", bound=generic)
+
+# Can't use the SupportsInt protocol here as the ndarray.__int__() method exists
```

Comment:
It might be worth considering removing ``__int__()``, ``__float__()`` and ``__complex__()`` from ``ndarray`` for now; at least until there is proper support for array shapes (as only 0D arrays actually support the use of these methods). 

This would most certainly make it easier to identify (numerical) scalars.

Reply:
Scratch this idea, the use of Protocols in identifying scalars might not be so straightforward as I initially thought (see https://github.com/numpy/numpy-stubs/pull/66#discussion_r412235311).